File: Easy/Java/28. Find the Index of the First Occurrence in a String.java
class Solution {
    public int strStr(String haystack, String needle) {
        int left = 0;
        int right = needle.length() - 1;

        while(right < haystack.length()){
            String temp = haystack.substring(left, right + 1);
            if(temp.equals(needle))
                return left;
            left++;
            right++;
        }
        
        return -1;
    }
}

File: Easy/JavaScript/28. Find the Index of the First Occurrence in a String.js
var strStr = function(haystack, needle) {
    let left = 0;
    let right = needle.length - 1;

    while(right < haystack.length){
        let temp = haystack.substring(left, right + 1);
        if(temp == needle)
            return left;
        left++;
        right++;
    }

    return -1;
};
